unison-share-api/src/Unison/Server/Endpoints/GetDefinitions.hs

```diff
@@ -105,36 +105,28 @@ instance ToSample DefinitionDisplayResults where
   toSamples _ = noSamples
 
 serveDefinitions ::
-  MonadIO m =>
   Rt.Runtime Symbol ->
   Codebase IO Symbol Ann ->
   Maybe ShortBranchHash ->
   Maybe NamespaceFQN ->
   [HashQualifiedName] ->
   Maybe Width ->
   Maybe Suffixify ->
-  Backend.Backend m DefinitionDisplayResults
+  Backend.Backend IO DefinitionDisplayResults
 serveDefinitions rt codebase mayRoot relativePath rawHqns width suff =
   do
     rel <-
       fmap Path.fromPath' <$> traverse (parsePath . Text.unpack) relativePath
-    ea <- liftIO . runExceptT $ do
-      root <- traverse (Backend.expandShortBranchHash codebase) mayRoot
-      let hqns = HQ.unsafeFromText <$> rawHqns
-          scope = case hqns of
-            -- TODO: Change this API to support being queried by just 1 name/hash
-            HQ.HashOnly _ : _ -> Backend.AllNames . fromMaybe Path.empty $ rel
```

Comment:
@hojberg  do you know what this block was for? I can likely maintain the current behaviour if necessary but I wasn't entirely sure of the purpose.
See the api transcript at the bottom of the PR to see what changed.

Reply:
We got it sorted out and adjusted the implementation and transcripts accordingly.